src/Orchard.Web/Modules/Orchard.OutputCache/Filters/CacheByRoleFilter.cs

```diff
@@ -85,25 +97,23 @@ public class CacheByRoleFilter : ICachingEventHandler {
                             (rp, p) => new UserPermission { RoleName = rp.Role.Name, PermissionName = p.FeatureName + "." + p.Name }
                         );
                 }
-            } else {
+            }
+            else {
                 // the anonymous user has no roles, get its permissions
-                permissionsQuery = GetPermissioFromRole("Anonymous");
+                permissionsQuery = GetPermissionsFromRole("Anonymous");
             }
 
             if (userRolesPermissionsQuery.Any()) {
                 userRolesPermissions.AddRange(userRolesPermissionsQuery
-                    .OrderBy(urp => urp.RoleName)
```

Comment:
I believe the order matter since it will change the key if the order changes.

Reply:
Roles and permissions are ordered a few rows afterwards.